chain/chain/src/store.rs

```diff
@@ -2278,23 +2285,30 @@ impl<'a> ChainStoreUpdate<'a> {
         self.merge(store_update);
     }
 
-    pub fn gc_outcomes(&mut self, block_hash: &CryptoHash) -> Result<(), Error> {
+    pub fn gc_outcomes(&mut self, block: &Block) -> Result<(), Error> {
+        let block_hash = block.hash();
         let mut store_update = self.store().store_update();
-        let outcome_ids = self.get_outcomes_by_block_hash(&block_hash)?;
-        for outcome_id in outcome_ids {
-            let mut outcomes_with_id = self.get_outcomes_by_id(&outcome_id)?;
-            outcomes_with_id.retain(|outcome| &outcome.block_hash != block_hash);
-            if outcomes_with_id.is_empty() {
-                self.gc_col(ColTransactionResult, &outcome_id.as_ref().into());
-            } else {
-                store_update.set_ser(
-                    ColTransactionResult,
-                    outcome_id.as_ref(),
-                    &outcomes_with_id,
-                )?;
+        for chunk_header in block.chunks().iter() {
+            if chunk_header.height_included() == block.header().height() {
```

Comment:
It seems to be a common use-case now. Should we consider adding a new method on the Block like `included_chunks`?

Reply:
It is not easy without cloning due to how `ShardChunkHeader` works